feat: discovery starting point model

xnought opened the pull request:
~Don't merge this yet, still refactoring and fixing.~ This is mergable now, check the comment below for more details

xnought commented:
<img width="1777" alt="Screen Shot 2022-10-10 at 6 18 26 PM" src="https://user-images.githubusercontent.com/65095341/194976162-9f869210-0453-4b0b-b5d3-80294befedf6.png">

# Summary
- Starting points show a scatterplot thumbnail of the overall scatter you are filtering from
- User can lasso a new starting point (saved only as ids)
- Users filter + project down from a given starting point

# Note
1. The filter + project mechanism is much better when you can view 2 scatterplots (starting and current)
2. BUT, its somewhat unintuitive for how the starting points relates to the left sidebar.
3. For clip, filter + project does absolutely nothing and not helpful

This was my hypothesis a couple months ago too. Perhaps the uncertainty in the embedding directly from the neural network that was trained is helpful for filter+reproject, but a default embedding (like clip) does not reveal anything new. Although clip is better at the start and for global exploration (not filter).

I'm struggling to justify filter + reproject. It was not helpful at all in my 10+ hour exploration of imagenet validation. I'm going to phase it out if no one can come up with a good reason why it should be used. + not interactive at all when running reprojection.

cabreraalex commented on frontend/src/discover/scatter/AutoScaleRegl.svelte:
```diff
@@ -2,13 +2,14 @@
+	export let data: ScatterRowsFormat = [];
 	export let colorRange: ColorRange = [];
 	export let config: ReglConfig = {};
+	export let resetSelection = true;
```
remove, not being used?

cabreraalex commented on frontend/src/discover/scatter/Scatter.svelte:
```diff
@@ -1,38 +1,54 @@
-	let height = 500;
-	$: width = innerWidth - 470;
+	let innerWidth: number, innerHeight: number;
+	export let width = 500;
```
move above with exports

cabreraalex commented on frontend/src/metadata/metadata.ts:
```diff
@@ -3,6 +3,7 @@ import type ColumnTable from "arquero/dist/types/table/column-table";
 import * as aq from "arquero";
 import {
+	interpolateYlGnBu,
 	interpolatePurples,
```
remove purples import

cabreraalex commented on zeno/mirror/sdm.py:
```diff
@@ -0,0 +1,24 @@
+EXAMPLE_SLICES = False
+
+
+def domino(cache_path, pd_with_preds_embed: pd.DataFrame, num_slices: int = 10):
```
So is this hardcoded currently?

cabreraalex replied:
(totally fine that it is, just curious)

xnought replied:
Yeah I'm running it in a colab notebook to experiment right now. And it actually has a gpu so it doesn't take 10 years on my laptop.

cabreraalex reviewed (changes requested):
The points on my screen are super small, not really usable. Can we change the scaling a bit?

xnought commented:
What do the points look like for you? For me they are pretty sizable and auto adjust based on the number of points showing.

xnought commented:
@cabreraalex Okay this is mergable now: I updated the point size and changed the color scale. I can't press the merge button, it's too much power. The ring will corrupt me.
![4k_fellowship_movie_screencaps com_5395](https://user-images.githubusercontent.com/65095341/195513644-9d32e230-f4fe-4b4a-b0ab-c6ed55719784.jpg)
